src/AltinnCore/Designer/Controllers/DeployController.cs

```diff
@@ -266,34 +267,33 @@ private async Task<bool> RegisterApplicationInStorage(string applicationOwnerId,
 
                     application.Title = applicationMetadataFromRepository.Title;
                     application.VersionId = versionId;
-                    if (application.Forms == null)
+                    if (application.ElementTypes == null)
                     {
-                        application.Forms = new List<ApplicationForm>();
+                        application.ElementTypes = new List<ElementType>();
                     }
 
-                    application.Forms = applicationMetadataFromRepository.Forms;
+                    application.ElementTypes = applicationMetadataFromRepository.ElementTypes;
 
                     HttpResponseMessage response = client.PutAsync(getApplicationMetadataUrl, application.AsJson()).Result;
                     if (response.IsSuccessStatusCode)
                     {
-                        _logger.LogInformation($"Application Metadata for {applicationId} is created. New versionId is {versionId}.");
+                        _logger.LogInformation($"Application Metadata for {appId} is created. New versionId is {versionId}.");
                     }
                     else
                     {
-                        _logger.LogInformation($"An error occured while trying to update application Metadata for {applicationId}. VersionId is {versionId}.");
+                        _logger.LogInformation($"An error occured while trying to update application Metadata for {appId}. VersionId is {versionId}.");
                     }
                 }
                 else if (getApplicationMetadataResponse.StatusCode == System.Net.HttpStatusCode.NotFound)
                 {
-                    ApplicationMetadata appMetadata = GetApplicationMetadata(applicationId, versionId);
-                    appMetadata.ApplicationOwnerId = applicationMetadataFromRepository.ApplicationOwnerId;
+                    Application appMetadata = GetApplicationMetadata(appId, versionId);
```

Comment:
is there a reasong the already created application object is not used? It is instanciated outside the if-loop and could be applied here.

Reply:
I removed appMetadata and reuse application